### ISSUE TITLE
uTox - portable mode option

### ISSUE THREAD

**Issue by darkman088** [GrayHatter/uTox#400]
Dear uTox developers,

I would kindly like to ask you, if it would be possible to add a
"Porable mode: Yes/No" combobox in the Profile tab in Settings ?
I hope, that this is a small request and doesn't require a lot of coding.

Many thanks in advance!

**Comment by Beast141** [comment]
Yes!! i agree.. this would be very useful and not needing to install it via ppa or anything!! for example like Bitmessage does, in the settings you can check 'portable mode' and it would move all files in its own folder, then you can move that folder on a USB stick.
and for updating the portable folder, you can do a git pull.

Using Ubuntu 16.04.
Is this feature difficult or a lot of work to do?

**Comment by GrayHatter** [comment]
I don't think it'll be too hard, the question is just how to default to portable mode without using the `-p` flag

uTox already can be started in portable mode by running `utox -p` from any terminal.

**Comment by Beast141** [comment]
You could add a button in the settings so that it persists with the -p flag.
and when you start it in portable mode with -p where is its folder with all settings and files and logs that you can move to a usb stick?

**Comment by GrayHatter** [comment]
@Beast141 the UI part is easy, I'm just trying to figure out how I can detect if it's should start in portable mode. Because it needs to read a file from somewhere, and just scanning the current directory doesn't work for me.

**Comment by Beast141** [comment]
okay. Also, is it possible to do a git pull for when you make it portable? So that it can easily update (every commit you do) in its own folder on a USB stick?

**Comment by GitHubHatesTor** [comment]
At least I've downloaded a zip one (utox_windows_x86-64), not a exe(installer), I was hoping uTox work as a portable mode.

**Comment by gaffatape** [comment]
@GrayHatter
That's going to be an oddball solution, if it can't check for a file in the same folder (could just be an empty dot-file). Any sane reason? ;)

A solution could be to rename utox to utoxp, when in portable mode and vice versa. As I'm guessing there isn't going to be any broken links if it's in portable mode anyway.

**Comment by GrayHatter** [comment]
I like that solution a lot actually. if the name (without extension) isn't `utox`, start in portable mode.

**Comment by cebe** [comment]
we could build a separate binary. utox and utox_portable.

**Comment by 2SI3NX** [comment]
@GrayHatter any updates?
